libenv/sysinfo.c

```diff
@@ -1554,7 +1560,18 @@ static void OSClasses(EvalContext *ctx)
         EvalContextClassPutHard(ctx, "unknown_ostype", "source=agent,derived-from=sys.sysname");
     }
 
-    SetFlavor(ctx, "windows");
+    char *release = SafeStringDuplicate(VSYSNAME.release);
+    char *major = NULL;
+    FindNextInteger(release, &major);
+    if NULL_OR_EMPTY(major)
```

Comment:
```suggestion
    if (NULL_OR_EMPTY(major))
```

We are not savages ;)